client/src/components/ChatRoom.tsx

```diff
@@ -1,95 +1,165 @@
+  const navigate = useNavigate();
+  const {
+    socket,
+    username,
+    leaveRoom,
+    messages,
+    setMessages,
+    currentRoom,
+    sendMessage,
+    roomId,
+  } = useSockets();
+  const newMessageRef = useRef<HTMLInputElement>(null);
 
+  const handleChange = (e: any) => {
+    setValue(e.target.value);
+  };
 
-    const handleChange = (e: any) => {
-        setValue(e.target.value)
-    }
+  const handleSubmit = (e: any) => {
+    e.preventDefault();
+    console.log('socketid: ', socket.id);
+    //sendMessage(value);
+  };
+
+  function handleSendMessage(e: any) {
+    const message = newMessageRef.current?.value;
+
+    e.preventDefault();
+    // const message = value
 
-    const handleSubmit = (e: any) => {
-        e.preventDefault();
-        console.log('socketid: ', socket.id)
-        sendMessage(value);
+    if (!String(message).trim()) {
+      return;
     }
 
-    const handleOnLeave = () => {
-    leaveRoom()
-    navigate("/lobby");
+    socket.emit('SEND_ROOM_MESSAGE', { roomId, message, username });
+
+    setMessages([
+      ...messages,
+      {
+        username: 'You',
+        message,
+        time: moment().format(`HH:mm`),
+      },
+    ]);
+    if (newMessageRef && newMessageRef.current) {
+      newMessageRef.current.value = '';
     }
+    console.log(message);
+    console.log(messages);
+  }
 
-    return (
-        <div style={rootstyle}>
-            <div style={chatsDivStyle}>
-                {/* <ChatBubble /> */}
-                <p>{currentRoom}</p>
-            </div>
-            <form style={formStyle} onSubmit={handleSubmit} >
-                <input value={value} onChange={handleChange} style={inputStyle} type="text" placeholder="Join the conversation..."/>
-            </form>
-            <button style={buttonStyle} onClick={handleOnLeave}>Leave room</button>
+  const handleOnLeave = () => {
+    leaveRoom();
+    navigate('/lobby');
+  };
+  if (!roomId) {
+    return <div />;
+  }
+  return (
+    <div style={rootstyle}>
+      <div style={chatsDivStyle}>
+        {/* <ChatBubble /> */}
+        <p>{currentRoom}</p>
+      </div>
+      <div>
+        <div>
+          {messages.map(({ message, username, time }, index) => {
+            return (
+              <div key={index}>
+                <div key={index}>
+                  <span>
+                    {username} - {time}
+                  </span>
+                  <br />
+                  <span>{message}</span>
+                </div>
+              </div>
+            );
+          })}
+        </div>
+        <div>
+          <form style={formStyle} onSubmit={handleSubmit}>
+            <input
+              value={value}
+              onChange={handleChange}
+              style={inputStyle}
+              type="text"
+              placeholder="Join the conversation..."
+              ref={newMessageRef}
+            />
+            <button onClick={handleSendMessage}>SEND</button> 
```

Comment:
Call this function on "onSubmit" instead. You can remove the current "handleSubmit"